netket/graph/lattice.py

```diff
@@ -226,13 +283,25 @@ def __init__(
         new_nodes = {old_node: new_node for new_node, old_node in enumerate(old_nodes)}
         graph = _nx.relabel_nodes(graph, new_nodes)
 
+        self._coords = dicts_to_array(self._atoms, "r_coord")
+        self._lattice_dims = _np.expand_dims(self.extent, 1) * self.basis_vectors
+
         # Order node names
         nodes = sorted(graph.nodes())
         edges = list(graph.edges())
         graph = _nx.MultiGraph()
         graph.add_nodes_from(nodes)
         graph.add_edges_from(edges)
 
+        self._inv_dims = _np.linalg.inv(self._lattice_dims)
+        frac_positions = _np.matmul(self._coords, self._inv_dims) % 1
+        int_positions = _np.around(frac_positions * 10 ** tol_digits).astype(int) % (
+            10 ** tol_digits
+        )
```

Comment:
If I understand this correctly this is a poor's man fixed point decimal number implementaiton. Is that right?
I have the following comments:

Quoting from numpy documentation 
```python
np.around uses a fast but sometimes inexact algorithm to round floating-point datatypes. 
```
Can we have an *exact* algorithm? Or, if not, can you tell me that this won't introduce errors or bugs? Or, if not, can't we add an extra digit of precision and then throw it away just to be safe?
Is this maybe linked to your 1 % 1 error?

Regardless, a comment saying what you are doing for whoever reads the code in the future without access to the PR discussion is welcome.
Especfially why we are doing that.

Reply:
> If I understand this correctly this is a poor's man fixed point decimal number implementaiton. Is that right?

Yeah that’s correct. 

> Or, if not, can you tell me that this won't introduce errors or bugs?

I don’t think it matters. It seems like the numpy bug is related to producing extra digits which we truncate away immediately. 

> Is this maybe linked to your 1 % 1 error?

As for the 1.0 % 1.0 bug it’s similar to the 1.0-1.0 bug, i.e. it produces a qualitatively different result based on which one is larger

Reply:
> Quoting from numpy documentation
> 
> ```python
> np.around uses a fast but sometimes inexact algorithm to round floating-point datatypes. 
> ```

It also says that it always returns `np.true_divide(np.rint(a * 10**decimals), 10**decimals)`. Since we always use `decimals=0`, our `around` is just `rint` which doesn't come with a similar caveat in the docs.

Reply:
We might want to write `rint` though for peace of mind and making a bit clearer what we're doing